core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -87,6 +91,24 @@ angular.module('oppia').component('preferencesPage', {
         _saveDataItem('user_bio', userBio);
       };
 
+      ctrl.checkBioChanged = function() {
+        ctrl.allowreload = false;
+      };
+
+      ctrl.saveBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
+          'beforeunload', ctrl.confirmBeforeLeaving);
+      };
+
+      ctrl.confirmBeforeLeaving = function(e) {
+        if (!ctrl.allowreload) {
+          // This message is irrelevant, but is needed to trigger the
+          // confirmation before leaving.
+          e.returnValue = 'Sure?';
+          return false;
```

Comment:
Why `false`? (I'm not sure what this value represents can you please explain? :) )

Reply:
TBH, I referred to #10584 and did the following changes. However, I'll look into the queries you have asked and respond.

Reply:
I removed the return stmt and it had no effect.

Reply:
Thanks for sharing the ref.

1. It's completely okay to copy a code (We all do this) but only if you completely understand how it works.
2. Will it be possible to place this code somewhere and use it in both places (to avoid duplicate code)? 

> I removed the return stmt and it had no effect.

I would suggest reading the ref doc and do as needed. [I'm not sure whether return in the event handler helps avoid propagation. You can ask the author of PR #10584 if needed.]